bridge.go

```diff
@@ -49,12 +54,32 @@ var _ http.Handler = &RequestBridge{}
 //
 // mcpProxy will be closed when the [RequestBridge] is closed.
 func NewRequestBridge(ctx context.Context, providers []Provider, recorder Recorder, mcpProxy mcp.ServerProxier, logger slog.Logger, metrics *Metrics, tracer trace.Tracer) (*RequestBridge, error) {
+	return NewRequestBridgeWithCircuitBreaker(ctx, providers, recorder, mcpProxy, logger, metrics, tracer, DefaultCircuitBreakerConfig())
```

Comment:
This doesn't give us any mechanism to configure the CB going forward.
In fact, I don't think a global CB is the right strategy here. Having per-provider CBs is probably more future-proof since different providers will exhibit different behaviours.

> that couldn't be implemented as generic HTTP middleware in coderd (since it requires understanding upstream provider responses).

This would also solve this problem, since you could have provider-specific CBs.